gateway/server.go

```diff
@@ -993,6 +995,30 @@ func (gw *Gateway) DoReload() {
 	mainLog.Info("API reload complete")
 }
 
+func (gw *Gateway) syncResourcesWithReload(resource string, syncFunc func() (int, error)) (int, error) {
+	var (
+		err   error
+		count int
+	)
+
+	if resource != "apis" && resource != "policies" {
+		return 0, ErrSyncResourceNotKnown
+	}
+
+	conf := gw.GetConfig()
+	for i := 1; i <= conf.ResourceReloadAttempts+1; i++ {
+		count, err = syncFunc()
+		if err == nil {
+			return count, nil
+		}
+
+		mainLog.Errorf("Error during syncing %s: %s, attempt count %d", resource, err.Error(), i)
```

Comment:
maybe skip the log here, log outside

Reply:
I feel like attempt count must be logged. wdyt?